lib/widgets/content.dart

```diff
@@ -839,6 +840,94 @@ class MathBlock extends StatelessWidget {
   }
 }
 
+class MessageLinkPreview extends StatelessWidget {
+  const MessageLinkPreview({super.key, required this.node});
+
+  final LinkPreviewNode node;
+
+  @override
+  Widget build(BuildContext context) {
+    final messageListTheme = MessageListTheme.of(context);
+    final isSmallWidth = MediaQuery.sizeOf(context).width <= 576;
+
+    final titleAndDescription = Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        if (node.title != null)
+          GestureDetector(
+            onTap: () => _launchUrl(context, node.hrefUrl),
+            child: Text(node.title!,
+              style: TextStyle(
+                fontSize: 1.2 * kBaseFontSize,
+                height: 1.0,
+                // Web has the same color in light and dark mode.
+                color: const HSLColor.fromAHSL(1, 200, 1, 0.4).toColor()))),
+        if (node.description != null)
+          Container(
+            padding: const EdgeInsets.only(top: 3),
+            constraints: const BoxConstraints(maxWidth: 500),
+            child: Text(node.description!,
+              style: const TextStyle(height: 1.4))),
+      ]);
+
+    final clippedTitleAndDescription = Container(
+      constraints: const BoxConstraints(maxHeight: 80),
+      padding: const EdgeInsets.symmetric(horizontal: 5),
+      child: InsetShadowBox(
+        bottom: 8,
+        // TODO(#647) use different color for highlighted messages
+        // TODO(#681) use different color for DM messages
+        color: messageListTheme.streamMessageBgDefault,
+        child: UnconstrainedBox(
+          alignment: Alignment.topLeft,
+          constrainedAxis: Axis.horizontal,
+          clipBehavior: Clip.antiAlias,
+          child: Padding(
+            padding: const EdgeInsets.only(bottom: 8.0),
+            child: isSmallWidth
+              ? titleAndDescription
+              : LayoutBuilder(
+                  builder: (context, constraints) => ConstrainedBox(
+                    constraints: BoxConstraints(
+                      maxWidth: constraints.maxWidth - 115),
```

Comment:
I wonder about dropping this `maxWidth - 115` detail, for a few benefits:

- More of the text can show before it gets clipped
- Removes the need for `LayoutBuilder` which isn't great for
  - performance
  - code complexity (e.g. my difficulty in a previous comment)
- We allow more horizontal space for other paragraph content without issues

Could leave a code comment saying we're not following web in this way.